app/tasks/maintenance_tasks/task.rb

```diff
@@ -74,6 +83,44 @@ def load_constants
       end
     end
 
+    # @api private
+    ActiveRecordEnumeratorBuilder = Struct.new(:relation) do
+      def enumerator(context:)
+        JobIteration::EnumeratorBuilder.new(nil).active_record_on_records(
+          relation,
+          cursor: context.cursor,
+        )
+      end
+    end
+    private_constant :ActiveRecordEnumeratorBuilder
+
+    # @api private
+    ArrayEnumeratorBuilder = Struct.new(:array) do
+      def enumerator(context:)
+        JobIteration::EnumeratorBuilder.new(nil).build_array_enumerator(
+          array,
+          cursor: context.cursor,
+        )
+      end
+    end
+    private_constant :ArrayEnumeratorBuilder
+
+    def enumerator_builder
+      collection = self.collection
+
+      case collection
+      when ActiveRecord::Relation
+        ActiveRecordEnumeratorBuilder.new(collection)
+      when Array
+        ArrayEnumeratorBuilder.new(collection)
+      else
+        raise ArgumentError, "#{self.class.name}#collection must be either "\
+          'an Active Record Relation, or Array.' # TODO: update
+        # If you want CSVs, do this
+        # If you want custom enum, do this
```

Comment:
I'm actually inclined to leave the error message exactly as you have it `("#{self.class.name}#collection must be either an Active Record Relation or Array."`) . Since neither CSV Tasks nor custom enumeration Tasks require a `#collection` method to be filled in, I think it's reasonable for the error to reference the only two types of objects that `collection` actually supports.

Maybe we can add some additional wording in the Task generator to help folks understand the different types of Tasks available to them instead.